packages/rule-https-only/src/rule.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview sonarwhal rule that checks if your site is using https and if it has Mixed Content.
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+
+import { IRule, RuleMetadata, Event, IAsyncHTMLElement } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+import { isHTTPS } from 'sonarwhal/dist/src/lib/utils/misc';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class HttpsOnlyRule implements IRule {
+    private targetHTTPs: boolean = false;
+    private targetResource: string = '';
+
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.security,
+            description: `sonarwhal rule that checks if your site is using https and if it has Mixed Content.`
+        },
+        id: 'https-only',
+        schema: [
+            /*
+             * If you want to allow the user to configure your rule
+             * you should use a valid JSON schema. More info in:
+             * https://sonarwhal.com/docs/contributor-guide/rules/#themetaproperty
+             */
+        ],
+        scope: RuleScope.site
+    }
+
+    public constructor(context: RuleContext) {
+
+        const validateTarget = async (event: Event): Promise<boolean> => {
+            const { resource } = event;
+
+            this.targetResource = resource;
+
+            if (!isHTTPS(resource)) {
+                debug('HTTPS no detected');
+
+                await context.report(resource, null, 'The site should be https');
+
+                return false;
+            }
+
+            debug('HTTPS detected');
+
+            this.targetHTTPs = true;
+
+            return true;
+        };
+
+        const validate = async (event: Event, eventName: string) => {
+            if (eventName === 'fetch::end::html') {
+                await validateTarget(event);
+
+                return;
+            }
+
+            if (!this.targetHTTPs) {
+                return;
+            }
+
+            const { resource } = event;
+
+            if (resource === this.targetResource) {
+                return;
+            }
+
+            if (!isHTTPS(resource)) {
+                await context.report(resource, null, 'Requested insecure content');
```

Comment:
Why not just used `Should be served over HTTPS`?

Reply:
I like it (I'm not very good for messages heheh)